src/autorest-core/lib/data-store/data-store.ts

```diff
@@ -320,8 +321,28 @@ export class DataStore extends DataStoreView {
     return Object.getOwnPropertyNames(this.store);
   }
 
-  public Blame(absoluteUri: string, position: SmartPosition): BlameTree {
+  public Blame(absoluteUri: string, position: SmartPosition, message?: (m: Message) => void): BlameTree {
     const data = this.ReadStrictSync(absoluteUri);
+
+    // verify if the path exists in the document
+    if ('path' in position) {
+      let x = (<any>JSON.parse(data.ReadData()));
+      let keys = <string[]>(<any>position)['path'];
+      let validKeys = [];
+      let currObj = x;
+      for (let i = 0; i < keys.length; ++i) {
+        if (!(keys[i] in currObj)) {
+          if (message !== undefined) {
+            message({ Channel: Channel.Warning, Text: 'Could not find exact path: \'' + keys + '\'', Details: position });
+          }
```

Comment:
would throw an equivalent Error if `message` is `undefined` (for using `Blame` in a stricter setting)